pyproject.toml

```diff
@@ -41,10 +41,10 @@ nmdc-version = "nmdc_schema.nmdc_version:cli"
 nmdc-data = "nmdc_schema.nmdc_data:cli"
 
 [tool.poetry.dependencies]
-python = "^3.7"
+python = "^3.9"
```

Comment:
Since the Actions infrastructure runs the checks on Python 3.9, I thought it made sense to enforce the package requirements to have 3.9 and above as well.

Reply:
I might have made a separate PR as this is a separate concern but OK to merge in this one